Client/HttpCookieExtension.swift

```diff
@@ -0,0 +1,93 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import WebKit
+import Shared
+import Deferred
+
+private let log = Logger.browserLogger
+
+public extension HTTPCookie {
+    
+    static var locallySavedFile: String {
+        return "CookiesData.json"
+    }
+    
+    typealias Success = Bool
+    class func saveToDisk(_ filename: String = HTTPCookie.locallySavedFile, completion: ((Success) -> Void)? = nil) {
+        let cookieStore = WKWebsiteDataStore.default().httpCookieStore
+        
+        /* For reason unkown the callback to getAllCookies is not called, when the save is done from Settings.
+         A possibility is https://bugs.webkit.org/show_bug.cgi?id=188242
+         Even with the issue being fixed it still sometimes doesn't work.
+         The network process is in suspened maybe?
+         
+         And for some reason fetch cookie records preemptively guarantees this works.
+         Best guess is that fetching records brings network process to active.
+         
+         Same applies to setting cookies back below.
+         */
+        WKWebsiteDataStore.default().fetchDataRecords(ofTypes: [WKWebsiteDataTypeCookies]) { (_) in}
+        cookieStore.getAllCookies { (cookies) in
+            if let baseDir = FileManager.documentDirectoryURL {
```

Comment:
What about replacing it with `guard` to remove one level of indentation? And do the same in rest of the methods in the file